src/actions/actions.ts

```diff
@@ -511,6 +511,21 @@ export class ActionPayloadTypes {
         dataItems: PendingDataItem[]
     } = z
 
+    /**
+     * TODO
+     * 
+     * @param bcName
+     * @param dataItems
+     * @param assocValueKey
+     * @param selected
+     */
+    changeDataItemsFull: {
+        bcName: string,
+        dataItems: PendingDataItem[],
```

Comment:
* You do not need to pass the data in payload as it is [already in the store](https://github.com/tesler-platform/tesler-ui/blob/728d8b5ea2bc630e72bcc82462c9b79ae1db8f14/src/components/FullHierarchyTable/FullHierarchyTable.tsx#L230); moreover it will be memory consuming considering that you passing all items of a large table hierarchy.
* You do not need `bcName` as `widgetName` give you more options and `bcName` usually directly derived from widget
* And `assocValueKey` as it is already in the store ([store.view.popupData.assocValueKey](https://github.com/tesler-platform/tesler-ui/blob/728d8b5ea2bc630e72bcc82462c9b79ae1db8f14/src/components/widgets/AssocListPopup/AssocListPopup.tsx#L187))
* `selected` also can be deduced by reducer based on `pendingChanges` and `store.data[bcName]` lengths - if they do not match then you mark all items as selected, if they match then you mark all items as deselected.